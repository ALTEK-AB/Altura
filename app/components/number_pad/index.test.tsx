import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NumberPad from '.';

describe('<NumberPad />', () => {
  test('should render all buttons', () => {
    const onPressMock = jest.fn();
    const {getByTestId, getAllByTestId} = render(
      <NumberPad onPress={onPressMock} />,
    );
    const numberPadComponent = getByTestId('number-pad-component');
    const buttons = getAllByTestId(/^number-pad-button-/);
    expect(numberPadComponent).toBeDefined();
    expect(buttons).toHaveLength(12);
  });

  test('should call onPress when a button is pressed', () => {
    const onPressMock = jest.fn();
    const {getByTestId} = render(<NumberPad onPress={onPressMock} />);
    const button = getByTestId('number-pad-button-1');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalledTimes(1);
    expect(onPressMock).toHaveBeenCalledWith(1);
  });

  test('NumberPad renders correctly', () => {
    const onPressMock = jest.fn();
    const tree = renderer.create(<NumberPad onPress={onPressMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
