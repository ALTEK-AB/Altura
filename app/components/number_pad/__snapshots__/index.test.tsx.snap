// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberPad /> NumberPad renders correctly 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "marginHorizontal": 20,
      "width": "75%",
    }
  }
  testID="number-pad-component"
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-1"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-2"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-3"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        3
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-4"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        4
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-5"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        5
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-6"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        6
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-7"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        7
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-8"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        8
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-9"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        9
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-0"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        0
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#58D0EA",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
      testID="number-pad-button-<<"
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        &lt;&lt;
      </Text>
    </View>
  </View>
</View>
`;
